lib/verify/response.rb

```diff
@@ -42,19 +42,25 @@ def scenario
     private
 
     def address
-      @address ||= parameters.fetch("attributes").fetch("addresses").first.fetch("value")
+      @address ||= most_recent_verified_value(parameters.fetch("attributes").fetch("addresses"))
     end
 
     def address_lines
       @address_lines ||= address.fetch("lines")
     end
 
     def full_name
-      first_name = parameters.fetch("attributes").fetch("firstNames").first.fetch("value")
-      middle_name = parameters.fetch("attributes").fetch("middleNames").first.fetch("value")
-      surname = parameters.fetch("attributes").fetch("surnames").first.fetch("value")
+      first_name = most_recent_verified_value(parameters.fetch("attributes").fetch("firstNames"))
+      middle_name = most_recent_verified_value(parameters.fetch("attributes").fetch("middleNames"))
+      surname = most_recent_verified_value(parameters.fetch("attributes").fetch("surnames"))
 
       [first_name, middle_name, surname].join(" ")
     end
+
+    def most_recent_verified_value(attributes)
+      attributes.sort_by { |attribute| Date.strptime(attribute["to"], "%Y-%m-%d") }
+        .select { |attribute| attribute["verified"] }
+        .last.fetch("value")
```

Comment:
What happens if there are no verified attributes?

Reply:
From the quote on the card it looks like we can always expect to retrieve one verified value.

> "unfortunately the rules are a bit weird on verified or not […] for names you should be ables to expect just one current verified value, but you may also get a current non-verified value. It may also be possible to get a current unverified name and only a verified previous name (possibly due to you recently changing your name)